crowbar_framework/lib/crowbar/error/upgrade.rb

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright 2017, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module Crowbar
+  module Error
+    # generic upgrade
+    class UpgradeError < StandardError
+    end
+
+    # cancel upgrade
+    class UpgradeCancelError < StandardError
+      def initialize(step_name = "")
+        super("Not possible to cancel the upgrade at the step #{step_name}")
```

Comment:
e.g. in case of the upgrade status errors we can't do that, as we require the library also from other places that don't have the translation rendering available from Rails. so i'd keep it non translated, at least for consistency with the other exceptions